src/index.ts

```diff
@@ -22,16 +25,84 @@ import {
   TransferRequestParams,
 } from './types'
 
+const NETWORK_CHAIN_IDS = {
+  alfajores: 44787,
+  mainnet: 42220,
+}
+
+const fetch = fetchCookie(nodeFetch)
+
 export default class FiatConnectClient implements FiatConectApiClient {
   config: FiatConnectClientConfig
+  signingFunction: (message: string) => Promise<string>
+  _sessionExpiry?: Date
 
-  constructor(config: FiatConnectClientConfig) {
+  constructor(
+    config: FiatConnectClientConfig,
+    signingFunction: (message: string) => Promise<string>,
+  ) {
     this.config = config
+    this.signingFunction = signingFunction
+  }
+
+  async _ensureLogin() {
+    const loginResult = await this.login()
+    if (!loginResult.ok) {
+      throw new Error(`Login failed: ${loginResult.val.error}`)
+    }
+  }
+
+  /**
+   * Logs in with the provider and initializes a session.
+   *
+   * @returns a Promise resolving to the literal string 'success' on a
+   * successful login or an Error response.
+   */
+  async login(): Promise<Result<string, ErrorResponse>> {
+    try {
+      if (this._sessionExpiry && this._sessionExpiry < new Date()) {
+        return Ok('success')
+      }
+      const expirationDate = new Date(Date.now() + 14400000) // 4 hours from now
+      const siweMessage = new SiweMessage({
+        domain: new URL(this.config.baseUrl).hostname,
+        address: this.config.accountAddress,
+        statement: 'Sign in with Ethereum',
+        uri: `${this.config.baseUrl}/auth/login`,
+        version: '1',
+        chainId: NETWORK_CHAIN_IDS[this.config.celoNetwork],
+        nonce: generateNonce(),
+        expirationTime: expirationDate.toISOString(),
+      })
+      const message = siweMessage.prepareMessage()
+      const body: AuthRequestBody = {
+        message,
+        signature: await this.signingFunction(message),
+      }
+
+      const response = await fetch(`${this.config.baseUrl}/auth/login`, {
```

Comment:
will this automagically add the session cookie to future fetch requests?

Reply:
yes, found this [fetch-cookie](https://www.npmjs.com/package/fetch-cookie) package today that takes of all this for us.